papers/extract.py

```diff
@@ -22,15 +20,40 @@
 # PDF parsing / crossref requests
 # ===============================
 
+def createrandname(path, ext):
+    import random
+    import string
+    check = False
+    while check == False:
+        rand = "".join(
+            random.sample(string.ascii_lowercase + string.digits, 20)
+        ) + ext
+        check = True if not os.path.exists(path + rand) else False
+    return rand
+
+
 def readpdf(pdf, first=None, last=None, keeptxt=False):
-    txtfile = pdf.replace('.pdf','.txt')
-    imgfile = pdf.replace('.pdf','.png')
+    import shutil
+    txtfile = pdf.replace('.pdf', '.txt')
+    imgfile = pdf.replace('.pdf', '.png')
     # txtfile = os.path.join(os.path.dirname(pdf), pdf.replace('.pdf','.txt'))
-    if True: #not os.path.exists(txtfile):
+    if True:  # not os.path.exists(txtfile):
         # logger.info(' '.join(['pdftotext','"'+pdf+'"', '"'+txtfile+'"']))
+        if os.path.isfile(pdf):
+            path, ext = os.path.splitext(pdf)
+            path = "/".join(path.split("/")[:-1])
```

Comment:
Regarding the dependencies, that's true, it's because of tesseract-ocr, the only other alternative would be to use something like PyPDF to read the pdf's and extract the text but still that I think that doesn't cover the case of scanned pdf's although can't say that 100% confident.

Reply:
I see. Given the sort of improvement you report, this should come as a necessary evil, until we find an alternative.